### DISCUSSION TITLE
magnetic-one: how to intercept messages sent among agents to display in user application (web app, etc.)?

### DISCUSSION THREAD

**Discussion by michalmar** [microsoft/autogen#4188] [Q&A] [answered]
I am planning to run agentic workflows based on magnetic-one and these will be started from web UI. I want to display the messages that are being sent from/to agents inline in the application to monitor status. What is the best approach?

So far I am evaluating:
- connect on `LogHandler` and display those logs (as messages)
- override/add some `callback` function (haven't found a way on how to do it)

**Comment by ekzhu** [comment]
You can wait for our effort on magentic one on agentchat #3756 

AgentChat's API (preview): https://microsoft.github.io/autogen/dev/user-guide/agentchat-user-guide/index.html

**Reply by michalmar** [reply]
@ekzhu  I will check that for sure, but meantime - is there a way how to achieve that in current non-chat implementation?

**Reply by ekzhu** [reply]
I believe at this point you need to look into the current magentic one implementation based on the Core API and emit messages to a topic that is subscribed to by a closure agent embedded in your application.

**Comment by husseinmozannar** [comment, marked as answer]
I recommend you look at the preview interface here https://github.com/microsoft/autogen/tree/main/python/packages/autogen-magentic-one/interface for ideas and maybe you can use it directly.

As Eric mentioned, things will change very soon with the agentchat implementation.
